FV3/makefile

```diff
@@ -24,8 +24,45 @@ PPSER_FLAGS = --verbose --ignore-identical -m utils_ppser_kbuff
 FV3_EXE  = fv3.exe
 FV3CAP_LIB  = libfv3cap.a
 
+PWD :=$(shell pwd)
+TOP_LEVEL_OBJS_RELATIVE = atmos_model.o module_fv3_config.o
+TOP_LEVEL_OBJS_ABSOLUTE = $(addprefix $(PWD)/, $(TOP_LEVEL_OBJS_RELATIVE))
+
+# Linking flags
+FV3_LIB_DIRS = atmos_cubed_sphere io ipd gfsphysics  ../stochastic_physics cpl coarse_graining
+FV3_LIBS_ABS = $(addprefix -L$(PWD)/, $(FV3_LIB_DIRS))
+FV3_LIB_FLAGS = -lfv3core -lfv3io -lipd -lgfsphys -lstochastic_physics -lfv3cpl -lfv3coarse_graining $(FV3_LIBS_ABS)
+
+ifeq ($(AI2_WRAPPER),Y)
+	MAYBE_WRAPPER_TARGET = wrapper_build
+endif
+
 all: libs fms_check
-	$(MAKE) $(FV3_EXE) $(MAKE_OPTS) FMS_DIR=$(FMS_DIR)
+	#  need to use this recursive makefile call to ensure that the libs are
+	#  built first
+	$(MAKE) $(FV3_EXE) $(MAYBE_WRAPPER_TARGET) $(MAKE_OPTS) FMS_DIR=$(FMS_DIR)
+
+
+# wrapper related makefile targets
+wrapper_build: $(TOP_LEVEL_OBJS_RELATIVE)
```

Comment:
We talked about how you'll add `all` to the rhs here and select this as the make target in the nix build.